src/web/session.py

```diff
@@ -242,17 +244,19 @@ def get_session_admin_players_filter_columns(
     def set_session_admin_players_filter_federations(
         cls, request: HTMXRequest, federations: list[Federation]
     ):
-        request.session[cls.ADMIN_PLAYERS_FILTER_FEDERATIONS_KEY] = federations
+        request.session[cls.ADMIN_PLAYERS_FILTER_FEDERATIONS_KEY] = [
+            federation.name for federation in federations
+        ]
 
     @classmethod
     def get_session_admin_players_filter_federations(
         cls, request: HTMXRequest
     ) -> list[Federation]:
-        # type-casting is needed because the value returned by Session.get is serialized
-        # when stored from a previous request (and kept as-is if stored by the current request)
         return [
-            d if isinstance(d, Federation) else Federation(d['name'])
-            for d in request.session.get(cls.ADMIN_PLAYERS_FILTER_FEDERATIONS_KEY, [])
+            Federation(federation)
+            for federation in request.session.get(
+                cls.ADMIN_PLAYERS_FILTER_FEDERATIONS_KEY, []
+            )
```

Comment:
Nice, maybe better with a different name:
```suggestion
            Federation(federation_name)
            for federation_name in request.session.get(
                cls.ADMIN_PLAYERS_FILTER_FEDERATIONS_KEY, []
            )
```